src/document-click.js

```diff
@@ -358,3 +347,52 @@ export function expandTargetHref_(e, target, urlReplacements) {
   }
   return newHref;
 }
+
+
+/** @typedef {{element: !Element, attributes: Object<string, string>}} */
+let RestoreElementAttributesDef;
+
+
+/**
+ * Removes attributes from elements with matching hash target. This includes:
+ *   *[id=hash] and a[name=hash].
+ * @param {!./service/ampdoc-impl.AmpDoc} ampdoc
+ * @param {string} hash
+ * @return {!Array<!RestoreElementAttributesDef>}}
+ * @private
+ */
+function removeAttrsWithMatchingHash_(ampdoc, hash) {
+  const restoreElementsAttrs = [];
+  const targetElements = ampdoc.getRootNode().querySelectorAll(
+      `#${hash},a[name=${hash}]`) || [];
```

Comment:
Used `dom.escapeCssSelectorIdent` to escape it. @dvoytenko is this accurate? Or if we should just use the `css-escape.cssEscape` instead?